frontend/model/contracts/group.js

```diff
@@ -1267,6 +1280,47 @@ sbp('chelonia/defineContract', {
             groupContractID: contractID,
             chatRoomID: data.chatRoomID
           })
+
+          // The following queued invocation handles reference counting. It uses
+          // some transient (non-persistent) state to see whether the
+          // pairing `retain` to `release` was called. This situation could
+          // arise when syncing a group from scratch (e.g., from a new device)
+          // for chatrooms that have been joined and then left. In that case,
+          // there'll be no call to `retain` and therefore calling `release`
+          // is incorrect.
+          sbp('chelonia/queueInvocation', contractID, async () => {
+            // Has `retain` been skipped?
+            // Note: we check for 'skipped' instead of 'called' because
+            // side-effects cannot mutate state. The 'skipped' case can be
+            // checked without persistent storage, while the 'called' case
+            // would require some kind of persistent state (since the reference
+            // count is itself persisted)
+            const skippedRetain = sbp('okTurtles.data/get', `gi.contracts/group/chatroom-skipped-${contractID}-${data.chatRoomID}-${data.joinedHeight}`)
+            // There's no use of this transient value, so it's deleted to
+            // preserve memory
+            sbp('okTurtles.data/delete', `gi.contracts/group/chatroom-skipped-${contractID}-${data.chatRoomID}-${data.joinedHeight}`)
```

Comment:
I feel like this `gi.contracts/group/chatroom-skipped-${contractID}-${data.chatRoomID}-${data.joinedHeight}` value should be saved to a const instead of re-typed here just to be safe and also DRY things

Reply:
Sure